File: misc/making_anagrams.py


def charIndex(c):
    return ord(c) - ord('a');

def number_needed(str1, str2):

    letterCount = [0 for i in range(26)]

    for c in str1:
        letterCount[charIndex(c)] = letterCount[charIndex(c)] + 1

    for c in str2:
        letterCount[charIndex(c)] = letterCount[charIndex(c)] - 1

    answer=0
    for count in letterCount:
        answer += abs(count)

    return answer



print("abd-def =", number_needed("abd", "def"))
print("aba-radef =", number_needed("aba", "radef"))

File: misc/making_anagrams.cc
#include <iostream>
#include <string>
#include <array>


using namespace std;

int number_needed(string str1, string str2)
{
        std::array<int,26> countLetters {};

        for (char& c : str1) 
                countLetters[int(c-'a')]++;
        
        for (char& c : str2) 
                countLetters[int(c-'a')]--;

        int sum=0;
        for (auto& count: countLetters) {
                sum += abs(count);
        }

        return sum;

}


int main()
{
        cout << "abc-def " << number_needed("abc", "def") << endl;
        cout << "abac-acabdeef " << number_needed("abac", "acabdeef") << endl;
}
